tests/ast/data/DataFrame.join.inner.column.test

```diff
@@ -0,0 +1,17 @@
+## TEST CASE
+
+df1 = session.table("test_table")
+
+df2 = session.table("test_table")
+
+df3 = df1.join(df2, "num").select(df1.num.alias("n1"), df1.str, df2.str)
+
+## EXPECTED OUTPUT
+
+res1 = session.table("test_table")
+
+res2 = session.table("test_table")
+
+res3 = res1.join(res2, on=("num"), how="inner")
```

Comment:
We should avoid generating the parentheses around the `on` expression. Would you mind creating a task or just adding a TODO, please?

Reply:
Resolved in https://github.com/snowflakedb/snowflake/pull/197145